src/main/java/org/sagebionetworks/WorkflowOrchestrator.java

```diff
@@ -191,12 +192,18 @@ public WorkflowOrchestrator(SynapseClient synapse,
 			// get annotation for the CWL entry point.  Does the file exist?
 			Annotations annotations = synapse.getAnnotationsV2(entityId);
 			Map<String, AnnotationsValue> annotationsMap = annotations.getAnnotations();
-			if (annotationsMap==null) throw
-					new IllegalStateException("Expected string annotation called "+
-							ROOT_TEMPLATE_ANNOTATION_NAME+" for "+entityId+" but the entity has no string annotations.");
+			if ( annotationsMap == null || annotationsMap.isEmpty() ) {
+				throw new IllegalStateException("Expected annotation called "+
+							ROOT_TEMPLATE_ANNOTATION_NAME+" for "+entityId+" but the entity has null or empty annotation map.");
+			}
 			AnnotationsValue valueAnnotations = annotationsMap.get(ROOT_TEMPLATE_ANNOTATION_NAME);
-			if (valueAnnotations == null || valueAnnotations.getValue() == null || valueAnnotations.getValue().size() == 0 ) throw
-					new IllegalStateException(entityId+" has no AnnotationValue called "+ROOT_TEMPLATE_ANNOTATION_NAME);
+			if (valueAnnotations == null || valueAnnotations.getValue() == null
+					|| valueAnnotations.getValue().isEmpty()) {
+				throw new IllegalStateException(entityId + " has no annotation called " + ROOT_TEMPLATE_ANNOTATION_NAME);
+			}
+			if (!valueAnnotations.getType().equals(AnnotationsValueType.STRING)) {
```

Comment:
As a good convention is usually better to check the other way around: AnnotationsValueType.STRING.equals(something that might be null) to avoid potential null pointers